src/response.rs

```diff
@@ -136,9 +139,93 @@ impl Response {
             ext: Extensions::new(),
             peer_addr: None,
             local_addr: None,
+            // XXX(Jeremiah): should this be autogenerated on 4xx and 5xx codes?
+            error: None,
         }
     }
 
+    /// Create a new response from an `http_types::Error`.
+    ///
+    /// This will store the error in the `Response`, allowing it to later be
+    /// checked via `Response::error()`.
+    ///
+    /// If the `Error`'s status had a status code that was not a 5XX server
+    /// error code, the response body will be set to the error's message, and
+    /// the content-type header will be set to `http_types::mime::Plain`.
+    #[cfg(not(feature = "unstable"))]
+    pub fn from_error(error: Error) -> Self {
+        // Only send the message if it is a non-500 range error. All
+        // errors default to 500 by default, so sending the error
+        // body is opt-in at the call site.
+        let was_server_error = error.status().is_server_error();
+        let body = if !was_server_error {
+            Body::from_string(error.to_string())
+        } else {
+            Body::empty()
+        };
+
+        let (trailers_sender, trailers_receiver) = sync::channel(1);
+        let mut res = Self {
+            status: error.status(),
+            headers: Headers::new(),
+            version: None,
+            body,
+            trailers_sender: Some(trailers_sender),
+            trailers_receiver: Some(trailers_receiver),
+            ext: Extensions::new(),
+            peer_addr: None,
+            local_addr: None,
+            error: Some(error),
+        };
+        if !was_server_error {
+            res.set_content_type(mime::PLAIN);
```

Comment:
Does the body content type not get transferred correctly?

Reply:
Ah looks like it does actually, tide just did this unnecessarily.

Reply:
Err, nope. We don't use `set_body`, we are constructing the struct directly, so no logic is available. We could instead call `copy_content_type_from_body` but that seems like unnecessary overhead, since it will only ever be this anyways.